### PR TITLE
Reinstate `RichTextBox_Text_GetWithHandle_ReturnsExpected`

### LINKED ISSUE #3778: Cannot get the expected result when set "\r\n" to RichTextBox.Text property
* .NET Core Version: 
5.0.0-rc.1.20425.1

* Have you experienced this same bug with .NET Framework?: 
No

**More info:**

1. This issue does not occur in .NET Framework.
2. This issue does not occur in .NET Core 3.1 and .NET 5.0 Preview 8. It's a regression issue in 5.0.0-rc.1.20425.1.  
3. It maybe caused by this PR: [3726](https://github.com/dotnet/winforms/pull/3726)

**Problem description:**

Cannot get the expected result in RichTextBox.Text property when included the "\r\n" in it.
image

**Expected behavior:**
Should get the below value from RichTextBox.Text property. Like it works in .NET Framework and previous version of .NET Core SDK.
image

**Minimal repro:**

1. Launch the attached project in VS.
2. Run it and observe the value for richTextBox1.Text in debugging. 
[Issue1-TestApp.zip](https://github.com/dotnet/winforms/files/5127204/Issue1-TestApp.zip)

### LINKED ISSUE #3867: Random "RangeChecks instrumentation code detected an out of range array access" in Debug x86 tests
### Description

5.0-RC2 Windows Forms x86 Debug test appear to be randomly (yet persistently) crashing. E.g.
* https://dev.azure.com/dnceng/internal/_build/results?buildId=806500&view=results
* https://dev.azure.com/dnceng/internal/_build/results?buildId=806484&view=logs&j=7c8326b9-0a5f-532a-e6de-db8515c72d9a&t=99915513-7350-519e-b509-739b27f0b8ea&s=ff05ad62-bb9a-53b6-ce9f-72f329a63e7c
* https://dev.azure.com/dnceng/public/_build/results?buildId=806509&view=artifacts&type=publishedArtifacts

However upon re-run tests often succeed.

There's a memdump: https://dev.azure.com/dnceng/_apis/resources/Containers/5033456/Windows_NT_Windows%20Debug?itemPath=Windows_NT_Windows%20Debug%2Fdotnet.exe.49836.dmp which appear to have crashed with the following error:
```
Unhandled exception at 0x770B51D0 (ntdll.dll) in dotnet.exe.49836.dmp: RangeChecks instrumentation code detected an out of range array access.
```

Looking at the stack trace I'm struggling to form a hypothesis where the Windows Forms code may cause the buffer overrun:
```
     System.Windows.Forms.dll!System.Windows.Forms.NativeWindow.DefWndProc(ref System.Windows.Forms.Message m = {System.Windows.Forms.Message}) Line 515    C#    Symbols loaded.
     System.Windows.Forms.dll!System.Windows.Forms.Control.DefWndProc(ref System.Windows.Forms.Message m = {System.Windows.Forms.Message}) Line 5040    C#    Symbols loaded.
     System.Windows.Forms.dll!System.Windows.Forms.Control.WndProc(ref System.Windows.Forms.Message m = {System.Windows.Forms.Message}) Line 13032    C#    Symbols loaded.
     System.Windows.Forms.dll!System.Windows.Forms.TextBoxBase.WndProc(ref System.Windows.Forms.Message m = {System.Windows.Forms.Message}) Line 2204    C#    Symbols loaded.
     System.Windows.Forms.dll!System.Windows.Forms.RichTextBox.WndProc(ref System.Windows.Forms.Message m = {System.Windows.Forms.Message}) Line 3725    C#    Symbols loaded.
     System.Windows.Forms.dll!System.Windows.Forms.Control.ControlNativeWindow.OnMessage(ref System.Windows.Forms.Message m = {System.Windows.Forms.Message}) Line 67    C#    Symbols loaded.
     System.Windows.Forms.dll!System.Windows.Forms.Control.ControlNativeWindow.WndProc(ref System.Windows.Forms.Message m = {System.Windows.Forms.Message}) Line 119    C#    Symbols loaded.
     System.Windows.Forms.dll!System.Windows.Forms.NativeWindow.Callback(System.IntPtr hWnd = 0x02a504ec, Interop.User32.WM msg = NCCREATE, System.IntPtr wparam = 0x00000000, System.IntPtr lparam = 0x13a9e0f0) Line 372    C#    Symbols loaded.
     System.Windows.Forms.dll!System.Windows.Forms.NativeWindow.WindowClass.Callback(System.IntPtr hWnd = 0x02a504ec, Interop.User32.WM msg = NCCREATE, System.IntPtr wparam = 0x00000000, System.IntPtr lparam = 0x13a9e0f0) Line 55    C#    Symbols loaded.
     [Native to Managed Transition]        Annotated Frame
     [Managed to Native Transition]        Annotated Frame
     System.Windows.Forms.Primitives.dll!Interop.User32.CreateWindowExW(Interop.User32.WS_EX dwExStyle = CLIENTEDGE, string lpClassName = "WindowsForms10.RichEdit20W.app.0.358a177_r3_ad1", string lpWindowName = null, Interop.User32.WS dwStyle = 1445003332, int X = 0, int Y = 0, int nWidth = 100, int nHeight = 96, System.IntPtr hWndParent = 0x01db0510, System.IntPtr hMenu = 0x00000000, System.IntPtr hInst = 0x00fb0000, object lpParam = null) Line 43    C#    Symbols loaded.
     System.Windows.Forms.dll!System.Windows.Forms.NativeWindow.CreateHandle(System.Windows.Forms.CreateParams cp = {System.Windows.Forms.CreateParams}) Line 460    C#    Symbols loaded.
     System.Windows.Forms.dll!System.Windows.Forms.Control.CreateHandle() Line 4934    C#    Symbols loaded.
     System.Windows.Forms.dll!System.Windows.Forms.TextBoxBase.CreateHandle() Line 1374    C#    Symbols loaded.
     System.Windows.Forms.dll!System.Windows.Forms.Control.CreateControl(bool fIgnoreVisible = false) Line 4998    C#    Symbols loaded.
     System.Windows.Forms.dll!System.Windows.Forms.Control.CreateControl() Line 4960    C#    Symbols loaded.
>    System.Windows.Forms.Tests.dll!System.Windows.Forms.Tests.RichTextBoxTests.RichTextBox_CheckRichEditWithVersionCanCreateOldVersions() Line 10538    C#    Symbols loaded.
```

The failures started after https://github.com/dotnet/winforms/pull/3821 was merged, but looking at the code that reportedly causes the failure it's not obvious how it could cause the error:
https://github.com/dotnet/winforms/blob/release/5.0-rc2/src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/RichTextBoxTests.cs#L10538

### Configuration

coreclr.dll	x86\shared\Microsoft.NETCore.App\5.0.0-rc.2.20458.14\coreclr.dll @ 482494f9ecc9a630599278b80893841462c90a5b

### PR DESCRIPTION
**Pull request by RussKie**
Reinstate `RichTextBox_Text_GetWithHandle_ReturnsExpected`

Relates to #3778
Resolves #3867

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/3862)

### CODE REVIEW

**Comment by RussKie** [comment]
@JeremyKuhne looks like the new tests destabilise the build, observed multiple crashes, all except one in Debug x86 configuration. https://github.com/dotnet/runtime/issues/42003 contains more info.
I'm not quite sure what's in the test is causing the failure.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/RichTextBoxTests.cs [hunk 1]
```diff
@@ -6853,8 +6853,8 @@ public static IEnumerable<object[]> RichTextBox_Text_GetWithHandle_TestData()
 
         private const string SAME = "SAME";
 
-        //[WinFormsTheory]
-        //[MemberData(nameof(RichTextBox_Text_GetWithHandle_TestData))]
+        // NB: do not convert this into a theory as it will run hundreds of tests
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/RichTextBoxTests.cs:6856]: NB?

Comment (RussKie) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/RichTextBoxTests.cs:6856]: "Nota Bene", or "note well". https://en.wikipedia.org/wiki/Nota_bene
I thought it was rather commonly used.

**Comment by RussKie** [comment]
Rebased on top of the latest master (which now has all 5.0 changes), so only updating the disabled test.

**Pull request merged**